finality-aleph/src/new_network/connection_manager/session.rs

```diff
@@ -30,10 +31,10 @@ pub enum AddressError {
 }
 
 async fn construct_authentication(
-    keychain: &KeyBox,
-    session_id: SessionId,
+    authority_index_and_pen: &Option<(NodeIndex, AuthorityPen)>,
+    session_id: Option<SessionId>,
```

Comment:
Pass this always (it makes sense always), just don't use it sometimes.

It should also be useful for returning the session id from the method below (I'm not sure, but it might be necessary even for non-validators; it certainly makes sense even for non-validators). You might want to avoid data duplication by making the field holding the authentication into some form of `Either` (perhaps custom defined?).

Reply:
Done